api/v1beta2/foundationdbcluster_types.go

```diff
@@ -328,6 +328,10 @@ type ProcessGroupStatus struct {
 	ProcessGroupConditions []*ProcessGroupCondition `json:"processGroupConditions,omitempty"`
 }
 
+// ProcessGroupID represents the ID of the process group
+// +kubebuilder:validation:MaxLength=1024
```

Comment:
I don't think length of 1024 is reasonable here, a realistic value is `63` based on https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set.